project/npda/models/patient.py

```diff
@@ -141,13 +142,31 @@ def age(self, today_date=None):
             today_date = self.get_todays_date()
         return stringify_time_elapsed(self.date_of_birth, today_date)
 
-    def save(self, *args, **kwargs) -> None:
+    async def asave(self, async_client, *args, **kwargs):
         if self.postcode:
+            try:
+                self.index_of_multiple_deprivation_quantile = await aimd_for_postcode(
+                    self.postcode, async_client
+                )
+            except HTTPError as err:
+                logger.warning(
+                    f"Cannot calculate deprivation score for {self.postcode} {err}"
+                )
+        
+        # HACK: as asave calls save internally and we don't want to call imd_for_postcode twice
```

Comment:
Maybe abandon this hack and just set imd in the calling code (patient form view and csv upload)? That's what I originally did in #200